src/HyperlaneTribunal.sol

```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.27;
+
+import {ITheCompactClaims} from "the-compact/src/interfaces/ITheCompactClaims.sol";
+import {ClaimWithWitness} from "the-compact/src/types/Claims.sol";
+import {Compact} from "the-compact/src/types/EIP712Types.sol";
+import {Tribunal} from "tribunal/Tribunal.sol";
+
+import {Router} from "hyperlane/contracts/client/Router.sol";
+
+string constant WITNESS_TYPESTRING =
+    "Mandate mandate)Mandate(uint256 chainId,address tribunal,address recipient,uint256 expires,address token,uint256 minimumAmount,uint256 baselinePriorityFee,uint256 scalingFactor,bytes32 salt)";
+
+library Message {
+    function encode(Tribunal.Compact calldata compact, bytes32 mandateHash, uint256 claimedAmount, address claimant)
+        internal
+        pure
+        returns (bytes memory)
+    {
+        require(
+            compact.allocatorSignature.length == 64 && compact.sponsorSignature.length == 64, "invalid signature length"
+        );
+
+        return abi.encodePacked(
+            compact.arbiter,
+            compact.sponsor,
+            compact.nonce,
+            compact.expires,
+            compact.id,
+            compact.maximumAmount,
+            compact.allocatorSignature,
+            compact.sponsorSignature,
+            mandateHash,
+            claimedAmount,
+            claimant
+        );
+    }
+
+    function decode(bytes calldata message)
+        internal
+        view
+        returns (
+            address sponsor,
+            uint256 nonce,
+            uint256 expires,
+            uint256 id,
+            uint256 allocatedAmount,
+            bytes calldata allocatorSignature,
+            bytes calldata sponsorSignature,
+            bytes32 witness,
+            uint256 claimedAmount,
+            address claimant
+        )
+    {
+        require(message.length == 380, "invalid message length");
+        address arbiter = address(bytes20(message[0:20]));
+        require(arbiter == address(this), "invalid arbiter");
```

Comment:
the mailbox will only call `handle` on the message headers recipient so this is unnecessary to include in the message body